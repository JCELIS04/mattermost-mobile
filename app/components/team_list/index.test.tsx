// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {fireEvent, renderWithEverything} from '@test/intl-test-helper';
import TestHelper from '@test/test_helper';

import TeamList from './index';

import type Database from '@nozbe/watermelondb/Database';
import type TeamModel from '@typings/database/models/servers/team';

describe('TeamList', () => {
    let database: Database;
    beforeAll(async () => {
        const server = await TestHelper.setupServerDatabase();
        database = server.database;
    });
    const teams = [
        {id: 'team1', displayName: 'Team 1'} as TeamModel,
        {id: 'team2', displayName: 'Team 2'} as TeamModel,
    ];
    beforeAll(async () => {
        const server = await TestHelper.setupServerDatabase();
        database = server.database;
    });

    it('should call onPress when a team is pressed', () => {
        const onPress = jest.fn();
        const {getByText} = renderWithEverything(
            <TeamList
                teams={teams}
                onPress={onPress}
            />,
            {database},
        );
        fireEvent.press(getByText('Team 1'));
        expect(onPress).toHaveBeenCalledWith('team1');
    });

    it('should render loading component when loading is true', () => {
        const {getByTestId} = renderWithEverything(
            <TeamList
                teams={teams}
                onPress={jest.fn()}
                loading={true}
            />,
            {database},
        );
        expect(getByTestId('team_list.loading')).toBeTruthy();
    });

    it('should render separator after the first item when separatorAfterFirstItem is true', () => {
        const {getByTestId} = renderWithEverything(
            <TeamList
                teams={teams}
                onPress={jest.fn()}
                separatorAfterFirstItem={true}
            />,
            {database},
        );
        expect(getByTestId('team_list.separator')).toBeTruthy();
    });
});
