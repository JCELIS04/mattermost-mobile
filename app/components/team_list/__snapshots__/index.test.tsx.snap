// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamList should render separator after the first item when separatorAfterFirstItem is true 1`] = `
<View
  style={
    {
      "flexGrow": 1,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      {
        "marginBottom": 4,
      }
    }
    data={
      [
        {
          "displayName": "Team 1",
          "id": "team1",
        },
        {
          "displayName": "Team 2",
          "id": "team2",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    handlerTag={4}
    handlerType="NativeViewGestureHandler"
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    testID="undefined.flat_list"
    viewabilityConfigCallbackPairs={[]}
    waitFor={
      [
        {
          "current": null,
        },
        {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 4,
              "display": "flex",
              "flexDirection": "row",
              "height": 56,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "height": 40,
                "width": 40,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1e325c",
                    "borderRadius": 8,
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                  {
                    "backgroundColor": "rgba(63,67,80,0.16)",
                  },
                ]
              }
              testID="team_sidebar.team_list.team_list_item.team1.team_icon"
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "textTransform": "uppercase",
                    },
                    {
                      "fontFamily": "Metropolis-SemiBold",
                      "fontSize": 20,
                      "fontWeight": "600",
                      "lineHeight": 28,
                    },
                    {
                      "color": "#3f4350",
                    },
                  ]
                }
                testID="team_sidebar.team_list.team_list_item.team1.team_icon.display_name_abbreviation"
              >
                Te
              </Text>
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "color": "#3f4350",
                  "flex": 1,
                  "fontFamily": "OpenSans",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                  "marginLeft": 16,
                },
                false,
              ]
            }
            testID="team_sidebar.team_list.team_list_item.team1.team_display_name"
          >
            Team 1
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "rgba(63,67,80,0.08)",
              "height": 1,
            }
          }
          testID="team_list.separator"
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 4,
              "display": "flex",
              "flexDirection": "row",
              "height": 56,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "height": 40,
                "width": 40,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1e325c",
                    "borderRadius": 8,
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                  {
                    "backgroundColor": "rgba(63,67,80,0.16)",
                  },
                ]
              }
              testID="team_sidebar.team_list.team_list_item.team2.team_icon"
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "textTransform": "uppercase",
                    },
                    {
                      "fontFamily": "Metropolis-SemiBold",
                      "fontSize": 20,
                      "fontWeight": "600",
                      "lineHeight": 28,
                    },
                    {
                      "color": "#3f4350",
                    },
                  ]
                }
                testID="team_sidebar.team_list.team_list_item.team2.team_icon.display_name_abbreviation"
              >
                Te
              </Text>
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "color": "#3f4350",
                  "flex": 1,
                  "fontFamily": "OpenSans",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                  "marginLeft": 16,
                },
                false,
              ]
            }
            testID="team_sidebar.team_list.team_list_item.team2.team_display_name"
          >
            Team 2
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
